### PR TITLE
Show LFC statistic in EXPLAIN

### PR DESCRIPTION
**Pull request by knizhnik**
## Problem

LFC has high impact on Neon application performance but there is no way for user to check efficiency of its usage

## Summary of changes

Show LFC statistic in EXPLAIN ANALYZE

## Description

**Local file cache (LFC)**

A layer of caching that stores frequently accessed data from the storage layer in the local memory of the Neon compute instance. This cache helps to reduce latency and improve query performance by minimizing the need to fetch data from the storage layer repeatedly.

**Externalization of LFC in explain output**

Then EXPLAIN ANALYZE output is extended to display important counts for local file cache (LFC) hits and misses.
This works both, for EXPLAIN text and json output.

**File cache: hits**

Whenever the Postgres backend retrieves a page/block from SGMR, it is not found in shared buffer but the page is already found in the LFC this counter is incremented.

**File cache: misses**

Whenever the Postgres backend retrieves a page/block from SGMR, it is not found in shared buffer and also not in then LFC but the page is retrieved from Neon storage (page server) this counter is incremented.

Example (for explain text output)

```sql
explain (analyze,buffers,prefetch,filecache) select count(*) from pgbench_accounts;
                                                                                         QUERY PLAN                                                                                         
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Finalize Aggregate  (cost=214486.94..214486.95 rows=1 width=8) (actual time=5195.378..5196.034 rows=1 loops=1)
   Buffers: shared hit=178875 read=143691 dirtied=128597 written=127346
   Prefetch: hits=0 misses=1865 expired=0 duplicates=0
   File cache: hits=141826 misses=1865
   ->  Gather  (cost=214486.73..214486.94 rows=2 width=8) (actual time=5195.366..5196.025 rows=3 loops=1)
         Workers Planned: 2
         Workers Launched: 2
         Buffers: shared hit=178875 read=143691 dirtied=128597 written=127346
         Prefetch: hits=0 misses=1865 expired=0 duplicates=0
         File cache: hits=141826 misses=1865
         ->  Partial Aggregate  (cost=213486.73..213486.74 rows=1 width=8) (actual time=5187.670..5187.670 rows=1 loops=3)
               Buffers: shared hit=178875 read=143691 dirtied=128597 written=127346
               Prefetch: hits=0 misses=1865 expired=0 duplicates=0
               File cache: hits=141826 misses=1865
               ->  Parallel Index Only Scan using pgbench_accounts_pkey on pgbench_accounts  (cost=0.43..203003.02 rows=4193481 width=0) (actual time=0.574..4928.995 rows=3333333 loops=3)
                     Heap Fetches: 3675286
                     Buffers: shared hit=178875 read=143691 dirtied=128597 written=127346
                     Prefetch: hits=0 misses=1865 expired=0 duplicates=0
                     File cache: hits=141826 misses=1865
```

The json output uses the following keys and provides integer values for those keys:

```
...
"File Cache Hits": 141826,
"File Cache Misses": 1865
...
```

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

**Review by Bodobolero — COMMENTED** [review]
Should we have a test case for this?
The test could verify:
- that the code works fine in Vanilla postgres (does not display the LFC stats)
- that the code works fine in Neon (does show the LFC stats) - I can help write the E-2-E test for Neon once this PR is merged similar to this one https://github.com/neondatabase/cloud/blob/4bcd22bbfc8ecf2bef068d4c10ba699dd9caa496/tests_e2e/tests/sql/neon_pg_stat_statements.sql 
https://github.com/neondatabase/cloud/blob/16fcc8fd6f8ca88aa01930f3bb85fa8c7f4960c3/tests_e2e/tests/v2_public_api_tests/test_e2e_sql_regress.py

Should we have documentation input for our writers that describes the information displayed?

**Review by lubennikovaav — CHANGES_REQUESTED** [review]
I approved postgres PRs
and this one needs rebase to bump postgres versions.
please ping me, to approve this one, when it is ready.

**Comment by knizhnik** [comment]
> I approved postgres PRs and this one needs rebase to bump postgres versions. please ping me, to approve this one, when it is ready.

Done

**Review by lubennikovaav — APPROVED** [review]

**Comment by Bodobolero** [comment]
For test case see https://github.com/neondatabase/neon/pull/6968

**Pull request merged**
